### PR TITLE
Improving config options for ntopng 4.2

### PR DESCRIPTION
**Pull request by mihakralj**
1. Fixing issues #2229 and #2096: allow multiple interfaces to be selected for ntopng
2. Adding  --disable-login (-l)  for ntopng to allow login-free access to ntopng GUI
3. Adding clarification text for httpport field: "Set to 0 to disable http listener."

### CODE REVIEW

File: net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf [hunk 1]
```diff
@@ -1,8 +1,13 @@
 {% if helpers.exists('OPNsense.ntopng.general.enabled') and OPNsense.ntopng.general.enabled == '1' %}
 {% from 'OPNsense/Macros/interface.macro' import physical_interface %}
 {%   if helpers.exists('OPNsense.ntopng.general.interface') and OPNsense.ntopng.general.interface != '' %}
--i={{ physical_interface(OPNsense.ntopng.general.interface) }}
+{%       for listen_interface in OPNsense.ntopng.general.interface.split(',') %}
+-i={{ physical_interface(listen_interface) }}
+{%       endfor %}
 {%   endif %}
+{%   if helpers.exists('OPNsense.ntopng.general.disablelogin') and OPNsense.ntopng.general.disablelogin != '' %}
```
Comment (AdSchellevis) [inline net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf:8]: we do have a shorthand for this nowadays, see the example below:
```
{%    if not helpers.empty('OPNsense.ntopng.general.disablelogin') %}
```

File: net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf [hunk 2]
```diff
@@ -1,8 +1,13 @@
 {% if helpers.exists('OPNsense.ntopng.general.enabled') and OPNsense.ntopng.general.enabled == '1' %}
 {% from 'OPNsense/Macros/interface.macro' import physical_interface %}
 {%   if helpers.exists('OPNsense.ntopng.general.interface') and OPNsense.ntopng.general.interface != '' %}
--i={{ physical_interface(OPNsense.ntopng.general.interface) }}
+{%       for listen_interface in OPNsense.ntopng.general.interface.split(',') %}
```
Comment (mimugmail) [inline net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf:4]: Indent always by 2

**Comment by mimugmail** [comment]
Can you bump version number in Makefile and add a changelog on pkg-desc? Also version in model should be bumped.

**Comment by mihakralj** [comment]
Major change from the previous pull request: I changed the HTTP field in the model from PortField to IntegerField - to allow 0 to be entered as a valid option. With HTTP port 0, ntopng disables HTTP listener and keep only https.

**Comment by fichtner** [comment]
Let‘s please not loose data models to expose configuration oddities.

**Comment by mihakralj** [comment]
I added constraints back into the model: integer only,  min 0, max 65535
If that's not acceptable, I'll return it back to PortField (which is actually integer, min1, max 65535(

**Comment by mihakralj** [comment]
Just checking if I have any pending actions here, or is this pull request cleared?

**Comment by mimugmail** [comment]
I tested the branch, but I'm unsure what the benefit of multiple -i is.
When I set LAN and WAN, I see both -i, but when I refresh the UI only see one selected, but nothing else interfaces in the dropdown.

Is this intended? THe rest looks gut, but please change version to 1.3 and remove revision

**Comment by mihakralj** [comment]
ntopng can restrict which interfaces are monitored by using multiple -i in its config file.
The expected result:
- if you select no interfaces in OPNsense plugin (no -i statements in ntopng.conf): ntopng provides ALL interfaces in the drop-down selection on top
- if you select desired interfaces in OPNsense plugin (one or more -i statements in ntopng.conf): ntopng will show only selected interfaces in the drop-down selection.

See snapshots: in OPNsense I chose only three interfaces for ntopng to monitor, there are only three available interfaces in ntopng UI

image
image

**Comment by mimugmail** [comment]
Ok, got it. I thought you were expecting when you select only LAN and WAN it will gather LAN and WAN combined. 
But if it's only for hiding it's ok. 

Maybe we should label it from Tracked Interfaces to Available Interfaces? Tracked would imply it already listens on them

**Comment by mihakralj** [comment]
Wording needs to imply that this selection will tell ntopng which subset of interfaces it will track - and will ignore all others. (btw, this is very important for free/community users as community edition limits the max number of interfaces it is willing to monitor - and this option helps users to choose interfaces that matter most)

I hoped that ntopng would give us helpful language:

    [—interface|-i] <interface|pcap>
The user can select one or more interfaces from the list so that ntopng will treat them as **monitored** interfaces. Any traffic flowing though monitored interfaces will be seen and processed by ntopng. The interface is passed using the interface number (e.g., -i 1) on Windows systems, whereas the name is used on Linux / Unix systems (e.g., -i eth0).

**Comment by mihakralj** [comment]
So, are we in agreement to change it to 'monitored interfaces'?

**Comment by mimugmail** [comment]
OK, great. Now only bump version and remove Revision as this is not only a fix :)

**Comment by mihakralj** [comment]
Before I do that... We are drifting apart with current ntop version and distribution - how do we reconsile that?

- the actual current ntopng version is 4.3 - far ahead from what is in OPNsense repo
    https://packages.ntop.org/FreeBSD/
- ntop team in Italy created their own .txz package for OPNsense, with their own UI
- their package is ALSO having the same flaws: no ability to down-select multiple interfaces, no option to bypass logon screen, no switch to force ntopng directly into community edition

This Pull Request could serve both sides - or even better, it should consolidate it all into a single package where OPNSense community can contribute to the OPNSense wrapper and ntop team can push the code forward.

Let me pull ntop engineers into this thread to consolidate.

**Comment by mimugmail** [comment]
Eh, no. OPNsense uses only upstream ports of BSD, ntop manage an own port which is binary-only, also not allowed here. You can only contribute to OPNsense plugin

**Comment by mihakralj** [comment]
ok, understood.

File: net/ntopng/pkg-descr [hunk 3]
```diff
@@ -8,6 +8,12 @@ and on Windows as well.
 Plugin Changelog
 ================
 
+1.2 r2
```
Comment (fichtner) [inline net/ntopng/pkg-descr:11]: It's better to make this 1.3, revision is only for fixes et al

File: net/ntopng/src/opnsense/mvc/app/models/OPNsense/Ntopng/General.xml [hunk 4]
```diff
@@ -7,13 +7,20 @@
...
+            <default>0</default>        
+            <Required>N</Required>
+        </disablelogin>         
         <interface type="InterfaceField">
             <Required>N</Required>
-            <Multiple>N</Multiple>
+            <Multiple>Y</Multiple>
         </interface>
-        <httpport type="PortField">
+        <httpport type="IntegerField">
```
Comment (fichtner) [inline net/ntopng/src/opnsense/mvc/app/models/OPNsense/Ntopng/General.xml:18]: I'm not merging this. Please add a checkbox..

File: net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf [hunk 5]
```diff
@@ -1,16 +1,21 @@
...
+{%   if not helpers.empty('OPNsense.ntopng.general.httpsport') %}
 -W={{ OPNsense.ntopng.general.httpsport }}
 {%   endif %}
-{%   if helpers.exists('OPNsense.ntopng.general.dnsmode') and OPNsense.ntopng.general.dnsmode != '' %}
+{%   if not helpers.empty('OPNsense.ntopng.general.dnsmode') %}
 -n={{ OPNsense.ntopng.general.dnsmode }}
 {%   endif %}
 -d=/var/db/ntopng
-{% endif %}
+{% endif %}
```
Comment (fichtner) [inline net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf:21]: Line ending mangled

File: net/ntopng/src/opnsense/mvc/app/models/OPNsense/Ntopng/General.xml (hunk 4 above)
Comment (mihakralj) [reply net/ntopng/src/opnsense/mvc/app/models/OPNsense/Ntopng/General.xml]: fixed - but not with yet another checkbox.
There is only one port in the model now, used for either HTTP (when no cert is selected) or for HTTPS traffic (when cert is selected). The unused option gets disabled in ntopng.conf

File: net/ntopng/pkg-descr (hunk 3 above)
Comment (mihakralj) [reply net/ntopng/pkg-descr]: done.

**Comment by mimugmail** [comment]
Dont get me wrong, I really appreciate your work and ideas, but when you add more features with every commit, instead of just fixing the requested stuff, we have to rereview from scratch which takes a bit time :)

**Comment by mihakralj** [comment]
Understood. Let me close this PR and come back once the modification is ready.

**Pull request closed without merging**